server/src/routes/profile/permissions.ts

```diff
@@ -161,32 +169,55 @@ export const getPermissions = ({ isAdmin, isProfileOwner, role, permissions }: P
     isCoreJsFeedbackVisible: false,
   };
 
-  return mapValues(defaultPermissions, (_, permission) =>
-    role &&
-    (([
-      'isStageInterviewFeedbackVisible',
-      'isStudentStatsVisible',
-      'isCoreJsFeedbackVisible',
-      'isProfileVisible',
-    ].includes(permission) &&
-      ['mentor', 'coursementor'].includes(role)) ||
-      ([
+  const accessToContacts = (permission: string, role: string = '') => {
+    return (
+      [
         'isEmailVisible',
         'isTelegramVisible',
         'isSkypeVisible',
         'isPhoneVisible',
         'isContactsNodesVisible',
         'isEnglishVisible',
-      ].includes(permission) &&
-        ['mentor'].includes(role)) ||
-      (['isProfileVisible'].includes(permission) && ['student'].includes(role)))
-      ? true
-      : (isProfileOwner && !['isStageInterviewFeedbackVisible', 'isCoreJsFeedbackVisible'].includes(permission)) ||
-        get(permissions, `${permission}.all`) ||
-        get(permissions, `${permission}.${role}`) ||
-        isAdmin ||
-        false,
-  );
+      ].includes(permission) && ['mentor', 'coursemanager'].includes(role)
+    );
+  };
+
+  const accessToFeedbacks = (permission: string, role: string = '') => {
+    return (
+      [
+        'isStageInterviewFeedbackVisible',
+        'isStudentStatsVisible',
+        'isCoreJsFeedbackVisible',
+        'isProfileVisible',
+      ].includes(permission) && ['mentor', 'coursementor', 'coursemanager'].includes(role)
+    );
+  };
+
+  const accessToProfile = (permission: string, role: string = '') =>
+    ['isProfileVisible'].includes(permission) && ['student'].includes(role);
+
+  return mapValues(defaultPermissions, (_, permission) => {
+    if (isAdmin || role === 'coursemanager') {
```

Comment:
i changed this function and made it more verbose but simpler for future modifications. initially it was hard to understand what condition does.